mreg_cli/api/history.py

```diff
@@ -63,12 +63,14 @@ class HistoryItem(BaseModel):
     data: dict[str, Any]
 
     @field_validator("data", mode="before")
-    def parse_json_data(cls, v: Any) -> dict[str, Any]:
-        """Ensure that data is always treated as a dictionary."""
+    def parse_json_data(cls, v: Any) -> Any:
+        """Ensure that non-dict values are treated as JSON."""
         if isinstance(v, dict):
-            return v  # type: ignore
-        else:
+            return v  # pyright: ignore[reportUnknownVariableType]
+        try:
             return json.loads(v)
+        except json.JSONDecodeError as e:
+            raise ValueError("Failed to parse history data as JSON") from e
```

Comment:
Pydantic will not handle failed JSON parsing in the validator gracefully, so we need to guard it. We should probably log this somehow.